src/main/java/org/prebid/server/bidder/adnuntius/AdnuntiusBidder.java

```diff
@@ -0,0 +1,252 @@
+        }
+
+        final List<AdnuntiusAdsUnit> adsUnits = bidResponse.getAdsUnits().stream()
+                .filter(AdnuntiusBidder::validateAdsUnit).collect(Collectors.toList());
+
+        if (adsUnits.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final String currency = extractCurrency(adsUnits);
+        return adsUnits.stream().map(adsUnit -> makeBid(adsUnit, currency)).collect(Collectors.toList());
+    }
+
+    private static boolean validateAdsUnit(AdnuntiusAdsUnit adsUnit) {
+        final List<AdnuntiusAd> ads = ObjectUtil.getIfNotNull(adsUnit, AdnuntiusAdsUnit::getAds);
+        return CollectionUtils.isNotEmpty(ads) && ads.get(0) != null;
+    }
+
+    private static String extractCurrency(List<AdnuntiusAdsUnit> adsUnits) {
+        final AdnuntiusBid bid = adsUnits.get(adsUnits.size() - 1).getAds().get(0).getBid();
+        return ObjectUtil.getIfNotNull(bid, AdnuntiusBid::getCurrency);
+    }
+
+    private static BidderBid makeBid(AdnuntiusAdsUnit adsUnit, String currency) throws PreBidException {
+        final AdnuntiusAd ad = adsUnit.getAds().get(0);
+        final String adId = ad.getAdId();
+        final Bid bid = Bid.builder()
+                .id(adId)
+                .impid(extractImpId(adsUnit))
+                .w(parseMeasure(ad.getCreativeWidth()))
+                .h(parseMeasure(ad.getCreativeHeight()))
+                .adid(adId)
+                .cid(ad.getLineItemId())
+                .crid(ad.getCreativeId())
+                .price(extractPrice(ad))
+                .adm(adsUnit.getHtml())
+                .adomain(extractDomain(ad.getDestinationUrls()))
+                .build();
+
+        return BidderBid.of(bid, BidType.banner, currency);
+    }
+
+    private static String extractImpId(AdnuntiusAdsUnit adsUnit) {
+        final String targetId = adsUnit.getTargetId();
+        final String auId = adsUnit.getAuId();
+
+        if (targetId == null || auId == null || !targetId.startsWith(auId + TARGET_ID_DELIMITER)) {
+            return null;
+        }
+
+        return targetId.substring(auId.length() + TARGET_ID_DELIMITER.length());
+    }
+
+    private static Integer parseMeasure(String measure) throws PreBidException {
+        try {
+            return Integer.valueOf(measure);
+        } catch (NumberFormatException e) {
+            throw new PreBidException(String.format("Value of measure: %s can not be parsed.", measure));
+        }
+    }
+
+    private static BigDecimal extractPrice(AdnuntiusAd ad) {
+        final BigDecimal amount = ObjectUtil.getIfNotNull(ad.getBid(), AdnuntiusBid::getAmount);
+        return amount != null ? amount.multiply(PRICE_MULTIPLIER) : BigDecimal.ZERO;
+    }
+
+    private static List<String> extractDomain(Map<String, String> destinationUrls) {
+        return destinationUrls == null ? Collections.emptyList() : destinationUrls.values().stream()
+                .map(url -> url.split("/")[2].replaceAll("www\\.", ""))
```

Comment:
Pls, check before accessing by index. I know that in go they ignore that check, but ArrayIndexOutOfBoundsException is ugly thing.